File: frontend/src/api/task.ts
import http from './http'

export interface Task {
  id: number
  pipelineId: number
  status: 'PENDING' | 'RUNNING' | 'SUCCESS' | 'FAILED' | 'CANCELLED'
  progress: number
  inputPath: string
  outputPath: string | null
  errorMsg: string | null
  createdAt: string
  finishedAt: string | null
}

export const taskApi = {
  submit: (pipelineId: number) =>
    http.post<Task>('/tasks', { pipelineId }),

  list: () => http.get<Task[]>('/tasks'),

  get: (id: number) => http.get<Task>(`/tasks/${id}`),

  cancel: (id: number) => http.post<Task>(`/tasks/${id}/cancel`),

  remove: (id: number) => http.delete<void>(`/tasks/${id}`),
}

File: backend/src/main/java/com/hades/dataflow/domain/entity/Task.java
package com.hades.dataflow.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("tasks")
public class Task {

    @Id
    private Long id;

    @Column("pipeline_id")
    private Long pipelineId;

    @Column("user_id")
    private Long userId;

    private String status;  // PENDING, RUNNING, SUCCESS, FAILED, CANCELLED

    private Integer progress;

    @Column("input_path")
    private String inputPath;

    @Column("output_path")
    private String outputPath;

    @Column("error_msg")
    private String errorMsg;

    @CreatedDate
    @Column("created_at")
    private LocalDateTime createdAt;

    @Column("finished_at")
    private LocalDateTime finishedAt;
}
